app/src/main/java/net/squanchy/support/lang/OptionalEnums.java

```diff
@@ -0,0 +1,16 @@
+package net.squanchy.support.lang;
+
+import android.text.TextUtils;
+
+import net.squanchy.eventdetails.domain.view.ExperienceLevel;
+
+public final class OptionalEnums {
+
+    public static Optional<ExperienceLevel> from(String rawLevel) {
```

Comment:
❌ why extracting this method into an util class? What did we say about not creating util classes? 😛 

This is a one-liner and you only use it twice... besides, `TextUtils.isEmpty()` is _definitely_ wrong, that would get you an `absent()` with a non-null input (the empty string)!

Please inline this (preferably), or make this a method on the `ExperienceLevel` itself, such as:

```java
public Optional<ExperienceLevel> fromNullableRawLevel(@Nullable String rawLevel) {
    if (rawLevel == null) {
        return Optional.absent();
    }
    return Optional.of(fromRawLevel(rawLevel));
}
```

---

Sigh I wish this Optional had `map`, it'd make this so much more compact (`Optional.ofNullable(rawLevel).map(() -> fromRawLevel(rawLevel))`)

@fourlastor I know you're against not using what's there already things but can we have our own less-barebones `Optional`? Please? :P